README.rst

```diff
@@ -117,107 +116,298 @@ The abstract command for running CALDP natively is:
+++++++++++++++++++++++++++++++++++++++++++++
+One technique for enabling AWS access inside the container is to put a *.aws* configuration directory in your
+*CALDP_HOME* directory.
+
+Since caldp-docker-run-pipeline mounts CALDP_HOME inside the container at *$HOME*, AWS will see them where it
+expects to find them.   AWS Batch nominally runs worker nodes which have the necessary permissions attached
+so no .aws directory is needed on AWS Batch.
+
+Output Structure
+----------------
+CALDP and CALCLOUD output data in a form desgined to help track the state of individual datasets.
+
+As such,  the output directory is organized into two subdirectories:
+
+1. *messages*
+2. *data*
+
+A key difference between CALDP and CALCLOUD is that the  former is designed for processing single
+datasets,  while the latter is designed for processing batches of datasets which are run individually
+by CALCLOUD.  In this context,  normally files downloaded from CALCLOUD's S3 storage to an onsite
+directory are placed in a "batch directory",  and the CALDP equivalent of that batch directory is
+the output directory.   The same messages and data appearing in the CALDP output directory would
+also appeaar in the sync'ed CALCLOUD batch directory.
+
+Messages Subdirectory
+=====================
+The *messages* subdirectory is used to record the status of individual datasets
+as they progress through processing, data transfer, and archiving.  Each dataset has a
+similarly named state file which moves between state directories as it starts or completes
+various states.   The dataset file can be used to record metadata but its primary use
+is to enable simple indentification dataset state without the use of a database, queues,
+etc.   Only a local file system is needed to track state using this scheme.   A mirror
+of this same scheme is used on the cloud on S3 storage to help guide file downloads from
+AWS.
+
+.. code-block:: sh
 
-Notably absent is `file:/home/einstein`,  mapping absolute paths into Docker hasn't been completely
-worked out yet and may never be feasible.
+    <output_path>/
+        messages/
+            datasets-processed/
+                <ipppssoots...>    # CALDP, normally running on AWS batch, leaves messages here.  they're empty.
+            dataset-synced/
+                <ipppssoots...>    # CALCLOUD's downloader leaves messages here, normally containing abspaths of files to archive.
+            dataset-archived/
+                <ipppssoots...>    # The archive can acknowledge archive completion here,  file contents should be preserved.
+
+Data Subdirectory
+=================
+The *data* subdirectory parallels but has a different structure than the *messages*
+subdirectory.   For every ipppssoot message,  there is a data directory and subdirectories
+which contain output files from processsing that ipppssoot.   In the current implementation,
+the ipppssoot message file is empty,  it is normally populated by the
```

Comment:
```populated by the``` what?!?!

Reply:
Damn.  Normally populated by calcloud.sync when it finishes downloading... with the abspaths of the files to archive.   That probably needs some work anyway but Lisa liked those.